### PR TITLE
[WIP] Switch chain button

### PR DESCRIPTION
**Pull request by solidovic**
### Description

Switch chain (to defaultChainId) button on wallet fallback.

merge after https://github.com/lidofinance/ethereum-staking-widget/pull/371

### Demo

Снимок экрана 2024-07-11 в 11 34 06

### Testing notes

Check all wallets;

Button use `defaultChainId` var.

### Checklist:

- [x] Checked the changes locally.
- [ ] Created / updated analytics events.
- [ ] Created / updated the technical documentation (README.md / [docs](https://docs.lido.fi/) / etc.).
- [ ] Affects / requires changes in other services (Matomo / Sentry / CloudFlare / etc.).

### CODE REVIEW

File: shared/wallet/l2-fallback/styles.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export const TextStyle = styled.p`
+  margin-bottom: 16px;
+`;
+
+export const ButtonStyle = styled((props) => <Button {...props} />)`
+  background: #ffffff1a;
+
+  &:not(:disabled):hover {
+    // TODO
```
Comment (Jeday) [inline shared/wallet/l2-fallback/styles.tsx:23]: What TODO?

File: shared/wallet/l2-fallback/l2-fallback.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export const L2Fallback: WalletCardComponent = (props) => {
+  const { chainId } = useAccount();
+  const { defaultChain } = useUserConfig();
+
+  const defaultChainName = useMemo(() => {
+    return CHAINS[defaultChain];
+  }, [defaultChain]);
+
+  const chainName = useMemo(() => {
+    return Object.keys(L2_CHAINS)[
```
Comment (Jeday) [inline shared/wallet/l2-fallback/l2-fallback.tsx:44]: Use object.entries to find by value and return key

Comment (Jeday) [reply shared/wallet/l2-fallback/l2-fallback.tsx:44]: Actually consider just reversing key and values in L2_CHAINS

File: shared/hooks/use-is-connected-wallet-and-supported-chain.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { useMemo } from 'react';
+import { useAccount } from 'wagmi';
+import { useUserConfig } from 'config/user-config';
+
+export const useIsConnectedWalletAndSupportedChain = () => {
```
Comment (Jeday) [inline shared/hooks/use-is-connected-wallet-and-supported-chain.ts:5]: Too long hook name, maybe useSupportedChainActive

File: features/withdrawals/claim/form/submit-button.tsx [hunk 4]
```diff
@@ -1,20 +1,28 @@
...
+import { FormatToken } from 'shared/formatters/format-token';
+import { useIsSupportedChain } from 'shared/hooks/use-is-supported-chain';
 import { isValidationErrorTypeUnhandled } from 'shared/hook-form/validation/validation-error';
 
+import { ClaimFormInputType, useClaimFormData } from '../claim-form-context';
+
 export const SubmitButton = () => {
-  const { active } = useWeb3();
+  const { isConnected } = useAccount();
+  const isSupportedChain = useIsSupportedChain();
```
Comment (Jeday) [inline features/withdrawals/claim/form/submit-button.tsx:15]: What if we combine all chain and account hooks into one?

Comment (Jeday) [reply features/withdrawals/claim/form/submit-button.tsx:15]: useConntectionStatus()

Comment (Jeday) [reply features/withdrawals/claim/form/submit-button.tsx:15]: isConnected // for just wallet connection to show or not connect button
isSupportedChain // to show UnsupportedChainButton
isChainL2 // inside UnsupportedChainButton
isActiveWallet -> isDappActive // to know that we can use our dapp features like stake

File: consts/chains.ts [hunk 5]
```diff
@@ -1,5 +1,15 @@
-export const enum CHAINS {
+export enum CHAINS {
   Mainnet = 1,
   Goerli = 5,
   Holesky = 17000,
 }
+
+export enum L2_CHAINS {
```
Comment (Jeday) [inline consts/chains.ts:7]: Reverse keys and values

**Review by Jeday — CHANGES_REQUESTED** [review]

File: shared/wallet/l2-fallback/styles.tsx (hunk 1 above)
Comment (solidovic) [reply shared/wallet/l2-fallback/styles.tsx:23]: we are waiting the design updates

File: shared/hooks/use-is-connected-wallet-and-supported-chain.ts (hunk 3 above)
Comment (solidovic) [reply shared/hooks/use-is-connected-wallet-and-supported-chain.ts:5]: A too-long hook name is not always bad.

The`useSupportedChainActive` isn't clearly.

File: features/withdrawals/claim/form/submit-button.tsx (hunk 4 above)
Comment (solidovic) [reply features/withdrawals/claim/form/submit-button.tsx:15]: 🤔

File: consts/chains.ts (hunk 5 above)
Comment (solidovic) [reply consts/chains.ts:7]: For what?

File: shared/wallet/l2-fallback/l2-fallback.tsx (hunk 2 above)
Comment (solidovic) [reply shared/wallet/l2-fallback/l2-fallback.tsx:44]: Is it optional?

Comment (Jeday) [reply shared/wallet/l2-fallback/l2-fallback.tsx:44]: No. If you are L2_CHAINS only to find keys by value you must reverse object.

File: features/withdrawals/claim/form/submit-button.tsx (hunk 4 above)
Comment (solidovic) [reply features/withdrawals/claim/form/submit-button.tsx:15]: good idea, let's leave it for refactoring (when we remove useWeb3)

**Pull request closed without merging**